### PR TITLE
Only run --upgrade on proxy upgrade less than Foreman 2.1

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: roles/foreman_installer/tasks/install.yml [hunk 1]
```diff
@@ -4,14 +4,17 @@
     msg: 'foreman_installer_options must be an array'
   when: foreman_installer_options is string
 
+- debug:
```
Comment (ekohl) [inline roles/foreman_installer/tasks/install.yml]: Is this still needed?

File: roles/foreman_installer/tasks/main.yml [hunk 2]
```diff
@@ -9,6 +9,9 @@
 - debug:
     msg: "Foreman installer version {{ foreman_installer_version_file['content'] | b64decode }}"
 
+- set_fact:
```
Comment (ekohl) [inline roles/foreman_installer/tasks/main.yml]: Perhaps this should be done before the debug. The debug can then use this variable.

File: roles/foreman_proxy_content/tasks/upgrade.yml [hunk 3]
```diff
@@ -1,16 +1,16 @@
 ---
 - include_tasks: certs_generate.yml
 
+- set_fact:
+    foreman_installer_options_internal_use_only:
```
Comment (ekohl) [inline roles/foreman_proxy_content/tasks/upgrade.yml]: This just wipes the internal options. The upgrade pipeline also sets these so I'm not sure that's wise.

**Comment by evgeni** [comment]
this slightly differs from the solution in #1143 ;)

**Pull request closed without merging**